sdks/python/apache_beam/transforms/window.py

```diff
@@ -295,19 +294,14 @@ def __lt__(self, other):
 class GlobalWindow(BoundedWindow):
   """The default window into which all data is placed (via GlobalWindows)."""
   _instance = None
-  # The maximum timestamp for global windows is MAX_TIMESTAMP - 1 day.
-  # This is due to timers triggering when the watermark passes the trigger
-  # time, which is only possible for timestamps < MAX_TIMESTAMP.
-  # See also GlobalWindow in the Java SDK.
-  _END_OF_GLOBAL_WINDOW = MAX_TIMESTAMP - (24 * 60 * 60)
 
   def __new__(cls):
     if cls._instance is None:
       cls._instance = super(GlobalWindow, cls).__new__(cls)
     return cls._instance
 
   def __init__(self):
-    super(GlobalWindow, self).__init__(GlobalWindow._END_OF_GLOBAL_WINDOW)
+    super(GlobalWindow, self).__init__(GlobalWindow._getTimestampFromProto())
```

Comment:
As a follow up change, should we cache this is a class level member instead of looking it up on each object?

Reply:
As far as I understand `GlobalWindow` is already a singleton and `__init__` will only be called once by the `__new__` method to initialize the `_instance` class member. Correct me if I'm wrong.

Reply:
Yes your right.